packages/odyssey-react-mui/src/RadioGroup.tsx

```diff
@@ -73,31 +89,79 @@ const RadioGroup = ({
   HintLinkComponent,
   id: idOverride,
   isDisabled,
+  isReadOnly = false,
   label,
   name: nameOverride,
   onChange: onChangeProp,
   testId,
   translate,
   value,
 }: RadioGroupProps) => {
+  const [internalValue, setInternalValue] = useState(value ?? defaultValue);
+
   const controlledStateRef = useRef(
     getControlState({
       controlledValue: value,
       uncontrolledValue: defaultValue,
     }),
   );
+
   const inputValues = useInputValues({
     defaultValue,
     value,
     controlState: controlledStateRef.current,
   });
 
+  const isControlled =
+    controlledStateRef.current === ComponentControlledState.CONTROLLED;
+
   const onChange = useCallback<NonNullable<MuiRadioGroupProps["onChange"]>>(
-    (event, value) => {
-      onChangeProp?.(event, value);
+    (event, newValue) => {
+      if (!isReadOnly) {
```

Comment:
I don't think we need to check for read only in the `onChange` handler. If we are handling the click correctly, the `onChange` won't ever fire right?

Reply:
This is needed because it prevents changing the selection if `isReadOnly` = true. It works that way because of the keyboard support for focusing, otherwise tabbing through the options of a Read-only radio would change the selection. 

I tested the behavior without and it allows selection change when Read-only.

Reply:
Per our Slack convo: Don't need the check for `isReadOnly` here as the change event is not fired due to the `preventDefault` in the `Radio` click handler